--- a/kabc/plugins/dir/resourcedirplugin.cpp
+++ b/kabc/plugins/dir/resourcedirplugin.cpp
@@ -23,7 +23,10 @@
 
 using namespace KABC;
 
-typedef KRES::PluginFactory<ResourceDir, ResourceDirConfig> ResourceDirFactory;
-
-K_EXPORT_PLUGIN(ResourceDirFactory)
-
+extern "C"
+{
+  KDE_EXPORT void *init_kabc_directory()
+  {
+    return new KRES::PluginFactory<ResourceDir, ResourceDirConfig>();
+  }
+}
--- a/kabc/plugins/file/resourcefileplugin.cpp
+++ b/kabc/plugins/file/resourcefileplugin.cpp
@@ -23,6 +23,10 @@
 
 using namespace KABC;
 
-typedef KRES::PluginFactory<ResourceFile, ResourceFileConfig> ResourceFileFactory;
-
-K_EXPORT_PLUGIN(ResourceFileFactory)
+extern "C"
+{
+  KDE_EXPORT void *init_kabc_file()
+  {
+    return new KRES::PluginFactory<ResourceFile, ResourceFileConfig>();
+  }
+}
--- a/kabc/plugins/ldapkio/resourceldapkioplugin.cpp
+++ b/kabc/plugins/ldapkio/resourceldapkioplugin.cpp
@@ -27,14 +27,11 @@
 
 using namespace KABC;
 
-class ResourceLDAPKIOFactory : public KRES::PluginFactory<ResourceLDAPKIO, ResourceLDAPKIOConfig>
+extern "C"
 {
-  public:
-    ResourceLDAPKIOFactory()
-      : KRES::PluginFactory<ResourceLDAPKIO, ResourceLDAPKIOConfig>()
-    {
-      KGlobal::locale()->insertCatalog( "kabc_ldapkio" );
-    }
-};
-
-K_EXPORT_PLUGIN(ResourceLDAPKIOFactory)
+  KDE_EXPORT void *init_kabc_ldapkio()
+  {
+    KGlobal::locale()->insertCatalog( "kabc_ldapkio" );
+    return new KRES::PluginFactory<ResourceLDAPKIO, ResourceLDAPKIOConfig>();
+  }
+}
--- a/kabc/plugins/net/resourcenetplugin.cpp
+++ b/kabc/plugins/net/resourcenetplugin.cpp
@@ -23,6 +23,10 @@
 
 using namespace KABC;
 
-typedef KRES::PluginFactory<ResourceNet, ResourceNetConfig> ResourceNetFactory;
-
-K_EXPORT_PLUGIN(ResourceNetFactory)
+extern "C"
+{
+  KDE_EXPORT void *init_kabc_net()
+  {
+    return new KRES::PluginFactory<ResourceNet, ResourceNetConfig>();
+  }
+}
