Author: José Manuel Santamaría Lema <panfaust@gmail.com>
Origin: http://commits.kde.org/kdepimlibs/614ec855f1b991927c586cf97526a787f2da4796
Description: Avoid ABI break compared with 4.8.x.
Applied-Upstream: 4.9.4

--- a/kabc/scripts/addressee.src.cpp
+++ b/kabc/scripts/addressee.src.cpp
@@ -544,7 +544,7 @@ void Addressee::removeKey( const Key &key )
   }
 }
 
-Key Addressee::key( Key::Type type, const QString& customTypeString ) const
+Key Addressee::key( Key::Type type, QString customTypeString ) const
 {
   Key::List::ConstIterator it;
   Key::List::ConstIterator end(d->mKeys.constEnd());
@@ -575,7 +575,7 @@ Key::List Addressee::keys() const
   return d->mKeys;
 }
 
-Key::List Addressee::keys( Key::Type type, const QString& customTypeString ) const
+Key::List Addressee::keys( Key::Type type, QString customTypeString ) const
 {
   Key::List list;
 
diff --git a/kabc/scripts/addressee.src.h b/kabc/scripts/addressee.src.h
index 90a858c..812efe0 100644
--- a/kabc/scripts/addressee.src.h
+++ b/kabc/scripts/addressee.src.h
@@ -276,7 +276,7 @@ class KABC_EXPORT Addressee
       @param customTypeString A string to match custom keys against when
              @p type is @c Key::Custom
      */
-    Key key( Key::Type type, const QString& customTypeString = QString() ) const;
+    Key key( Key::Type type, QString customTypeString = QString() ) const;
 
     /**
       Return list of all keys.
@@ -299,7 +299,7 @@ class KABC_EXPORT Addressee
       @param customTypeString A string to match custom keys against when
              @p type is @c Key::Custom
      */
-    Key::List keys( Key::Type type, const QString& customTypeString = QString()  ) const;
+    Key::List keys( Key::Type type, QString customTypeString = QString()  ) const;
 
     /**
       Return key with the given id.
-- 
1.7.10.4

