Index: kdepimlibs/kimap/tests/loginjobtest.cpp
===================================================================
--- kdepimlibs.orig/kimap/tests/loginjobtest.cpp	2014-05-18 00:31:24.641426654 +0200
+++ kdepimlibs/kimap/tests/loginjobtest.cpp	2014-05-18 00:31:58.984090428 +0200
@@ -226,7 +226,7 @@
             << "S: A000002 OK";
 
     QTest::newRow( "sslv3" ) << scenario << static_cast<int>(QSsl::SslV3) << static_cast<int>(KIMAP::LoginJob::SslV3);
-    QTest::newRow( "sslv2" ) << scenario << static_cast<int>(QSsl::SslV2) << static_cast<int>(KIMAP::LoginJob::SslV2);
+    // QTest::newRow( "sslv2" ) << scenario << static_cast<int>(QSsl::SslV2) << static_cast<int>(KIMAP::LoginJob::SslV2);
     //AnySslVersion doesn't mean the server can force a specific version (e.g. openssl always starts with a sslv2 hello)
     QTest::newRow( "any protocol with anyssl version" ) << scenario << static_cast<int>(QSsl::AnyProtocol) << static_cast<int>(KIMAP::LoginJob::AnySslVersion);
     //KIMAP and KTcpSocket use SslV3_1 but really mean tls without starttls
